Products/CMFPlone/browser/favicon.py

```diff
@@ -0,0 +1,26 @@
+from Products.CMFPlone.interfaces import ISiteSchema
+from plone.formwidget.namedfile.converter import b64decode_file
+from plone.namedfile.browser import Download
+from plone.namedfile.file import NamedImage
+from plone.registry.interfaces import IRegistry
+from zope.component import getUtility
+from plone.memoize import ram
+
+
+class SiteFavicon(Download):
+
+    def __init__(self, context, request):
+        super().__init__(context, request)
+        self.filename = None
+        self.data = None
+
+        registry = getUtility(IRegistry)
+        settings = registry.forInterface(ISiteSchema, prefix="plone")
+        if getattr(settings, 'site_favicon', False):
```

Comment:
What's the fallback when there is no favicon?  Should be included along the logo image, right?

Reply:
Why no fallback in this case ?
- If there is no favicon this view will not be called
- The Fallback is the Plone-default-favico at ".../Plonesite/favicon.ico" in that case that there is no favicon